### PR TITLE
feat(gitlab): support new backend system

### PR DESCRIPTION
**Pull request by zhammer**

### CODE REVIEW

File: plugins/gitlab-catalog-backend/README.md [hunk 1]
```diff
@@ -31,21 +31,12 @@ Enriches existing `User` entities with information from Gitlab, notably the user
...
-  );
-  const { processingEngine, router } = await builder.build();
-  processingEngine.start();
-  return router;
-}
+// in your imports
+import { catalogModuleGitlabUserProcessor } from '@seatgeek/backstage-plugin-gitlab-catalog-backend';
+
+// in your catalog modules
+backend.add(catalogModuleGitlabUserProcessor());
```
Comment (zhammer) [inline plugins/gitlab-catalog-backend/README.md:41]: once https://github.com/seatgeek/backstage-plugins/pull/61 is merged, confirm this works

File: plugins/gitlab-catalog-backend/src/GitlabUserProcessor.ts [hunk 2]
```diff
@@ -23,7 +23,7 @@ const GITLAB_DEFAULT_HOST = 'https://gitlab.com';
  */
 export class GitlabUserProcessor implements CatalogProcessor {
   private readonly gitlab: InstanceType<typeof Gitlab>;
-  private readonly logger: Logger;
```
Comment (zhammer) [inline plugins/gitlab-catalog-backend/src/GitlabUserProcessor.ts:26]: todo: note this as a breaking change in the release notes

(though it shouldn't cause too much toil as the `LoggerService` is an interface that seems to support the winston logger

**Review by bbckr — APPROVED** [review]
LGTM, would be nice to test this out on an actual backstage instance, but I think the local dev setup should be good enough for this.

Are there unit tests though around these plugins tho?

**Comment by zhammer** [comment]
@bbckr just ran this locally and confirmed the set up works.

we have unit tests on the gitlabuserprocessor!

**Pull request merged**